File: C++ solution code/check_two_string_are_k_anargram_or_not.cpp
#include<bits/stdc++.h>
using namespace std;

bool checkAnargram(string s1, string s2, int k)
{
    int len1 = s1.size(), len2 = s2.size();
    if(len1 != len2) return false;
    int data[256] = {0};

    for(int i = 0; i < len1; i++) {
        data[s1[i] - 'a']++;
    }

    int cnt = 0;
    for(int i = 0; i < len2; i++) {
        if(data[s2[i]-'a'] > 0) data[s2[i]-'a']--;
        else cnt++;
        if(cnt > k) return false;
    }

    return true;
}
int main()
{
    string s1, s2;
    int k;
    cin >> s1 >> s2 >> k;
    cout << (checkAnargram(s1, s2, k)?"YES":"NO") << endl;
    return 0;
}

File: python solution code/check_two_string_are_k_anargram_or_not.py

def anargramCheck(s1, s2, k):
    l1, l2 = len(s1), len(s2)
    if l1 != l2:
        return False
    data = [0] * 26
    cnt = 0

    for i in range(l1):
        data[ord(s1[i]) - ord('a')] += 1
    
    for i in range(l2):
        if(data[ord(s2[i]) - ord('a')] > 0):
            data[ord(s2[i]) - ord('a')] -= 1
        else:
            cnt += 1
        if( cnt > k):
            return False
    return True



s1 = input()
s2 = input()
k = int(input())

if(anargramCheck(s1, s2, k)):
    print("YES")
else:
    print("NO")